Mathlib/Analysis/Normed/Group/SemiNormedGroup.lean

```diff
@@ -0,0 +1,297 @@
+  coe X := X.α
+
+instance : LargeCategory.{u} SemiNormedGroup₁ where
+  Hom X Y := { f : NormedAddGroupHom X Y // f.NormNoninc }
+  id X := ⟨NormedAddGroupHom.id X, NormedAddGroupHom.NormNoninc.id⟩
+  comp {X Y Z} f g := ⟨g.1.comp f.1, g.2.comp f.2⟩
+
+/-- Porting Note: Added-/
+instance {X Y : SemiNormedGroup₁} : CoeFun (X ⟶ Y) fun _ => X → Y where
+  coe (f : X ⟶ Y) := f.1
+
+@[ext]
+theorem hom_ext {M N : SemiNormedGroup₁} (f g : M ⟶ N) (w : (f : M → N) = (g : M → N)) : f = g :=
+  Subtype.eq (NormedAddGroupHom.ext (congr_fun w))
+#align SemiNormedGroup₁.hom_ext SemiNormedGroup₁.hom_ext
+
+instance : ConcreteCategory.{u} SemiNormedGroup₁ where
+  forget :=
+    { obj := fun X => X
+      map := fun f => f }
+  forget_faithful := { }
+
+/-- Construct a bundled `SemiNormedGroup₁` from the underlying type and typeclass. -/
+def of (M : Type u) [SeminormedAddCommGroup M] : SemiNormedGroup₁ :=
+  Bundled.of M
+#align SemiNormedGroup₁.of SemiNormedGroup₁.of
+
+instance (M : SemiNormedGroup₁) : SeminormedAddCommGroup M :=
+  M.str
+
+/-- Promote a morphism in `SemiNormedGroup` to a morphism in `SemiNormedGroup₁`. -/
+def mkHom {M N : SemiNormedGroup} (f : M ⟶ N) (i : f.NormNoninc) :
+    SemiNormedGroup₁.of M ⟶ SemiNormedGroup₁.of N :=
+  ⟨f, i⟩
+#align SemiNormedGroup₁.mk_hom SemiNormedGroup₁.mkHom
+
+@[simp]
+theorem mkHom_apply {M N : SemiNormedGroup} (f : M ⟶ N) (i : f.NormNoninc) (x) :
+    mkHom f i x = f x :=
+  rfl
+#align SemiNormedGroup₁.mk_hom_apply SemiNormedGroup₁.mkHom_apply
+
+/-- Promote an isomorphism in `SemiNormedGroup` to an isomorphism in `SemiNormedGroup₁`. -/
+@[simps]
+def mkIso {M N : SemiNormedGroup} (f : M ≅ N) (i : f.hom.NormNoninc) (i' : f.inv.NormNoninc) :
+    SemiNormedGroup₁.of M ≅ SemiNormedGroup₁.of N where
+  hom := mkHom f.hom i
+  inv := mkHom f.inv i'
+  hom_inv_id := by
+    apply Subtype.eq
+    exact f.hom_inv_id
+  inv_hom_id := by
+    apply Subtype.eq
+    exact f.inv_hom_id
+#align SemiNormedGroup₁.mk_iso SemiNormedGroup₁.mkIso
+
+instance : HasForget₂ SemiNormedGroup₁ SemiNormedGroup
+    where forget₂ :=
+    { obj := fun X => X
+      map := fun f => f.1 }
+
+@[simp]
+theorem coe_of (V : Type u) [SeminormedAddCommGroup V] : (SemiNormedGroup₁.of V : Type u) = V :=
+  rfl
+#align SemiNormedGroup₁.coe_of SemiNormedGroup₁.coe_of
+
+@[simp]
+theorem coe_id (V : SemiNormedGroup₁) : ⇑(𝟙 V) = id :=
+  rfl
+#align SemiNormedGroup₁.coe_id SemiNormedGroup₁.coe_id
+
+-- Porting note : removed `simp` attribute for not being simp normal form
+theorem coe_comp {M N K : SemiNormedGroup₁} (f : M ⟶ N) (g : N ⟶ K) : (f ≫ g : M → K) = g ∘ f :=
+  rfl
+#align SemiNormedGroup₁.coe_comp SemiNormedGroup₁.coe_comp
+
+/--Porting Note: Added to make `coe_comp'` work -- might cause double coercions-/
+instance coeToNormedAddGroupHom {M N : SemiNormedGroup₁} : Coe (M ⟶ N) (NormedAddGroupHom M N) :=
+  ⟨fun f => f.1⟩
+
+-- If `coe_fn_coe_base` fires before `coe_comp`, `coe_comp'` puts us back in normal form.
```

Comment:
This comment no longer makes sense. At least a porting note saying so.

Reply:
I'm not sure what's going on here, but did as you suggested.